pkg/operator/pdf/v0/main.go

```diff
@@ -0,0 +1,89 @@
+//go:generate compogen readme --operator ./config ./README.mdx
+package pdf
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"sync"
+
+	_ "embed" // embed
+
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/instill-ai/component/pkg/base"
+)
+
+const (
+	taskConvertPdfToMarkdown string = "TASK_CONVERT_PDF_TO_MARKDOWN"
+)
+
+var (
+	//go:embed config/definition.json
+	definitionJSON []byte
+	//go:embed config/tasks.json
+	tasksJSON []byte
+	once      sync.Once
+	op        *operator
+)
+
+// Operator is the derived operator
+type operator struct {
+	base.Operator
+}
+
+// Execution is the derived execution
+type execution struct {
+	base.OperatorExecution
+}
+
+// Init initializes the operator
+func Init(bo base.Operator) *operator {
+	once.Do(func() {
+		op = &operator{Operator: bo}
+		err := op.LoadOperatorDefinition(definitionJSON, tasksJSON, nil)
+		if err != nil {
+			panic(err)
+		}
+	})
+	return op
+}
+
+func (o *operator) CreateExecution(sysVars map[string]any, task string) (*base.ExecutionWrapper, error) {
+	return &base.ExecutionWrapper{Execution: &execution{
+		OperatorExecution: base.OperatorExecution{Operator: o, SystemVariables: sysVars, Task: task},
+	}}, nil
+}
+
+// Execute executes the derived execution
+func (e *execution) Execute(_ context.Context, inputs []*structpb.Struct) ([]*structpb.Struct, error) {
+	outputs := []*structpb.Struct{}
+
+	for _, input := range inputs {
+		switch e.Task {
+		case taskConvertPdfToMarkdown:
+			inputStruct := ConvertPdfToMarkdownInput{}
+			err := base.ConvertFromStructpb(input, &inputStruct)
+			if err != nil {
+				return nil, err
+			}
+
+			scriptPath := "/component/pkg/operator/pdf/v0/python/pdfTransformer.py"
+			pythonInterpreter := "/opt/venv/bin/python"
+			cmd := exec.Command(pythonInterpreter, scriptPath)
+
+			outputStruct, err := convertPdfToMarkdown(inputStruct, cmd)
```

Comment:
Conceptually, the command variables and executor initialisation belong in the `convert` method. `Execute` should only be responsible of transforming from / to structpb and multiplexing the logic to each task. Initialising the command outside of the method doesn't have any benefit here.

```suggestion
			outputStruct, err := convertPdfToMarkdown(inputStruct)
```

Reply:
Thanks for your feedback!

> However, if possible we should avoid modifying the production code just because it's more convenient for testing*.

I also agree on that! It is also a hard decision for me.
In Ruby, we can easily mock command without dependency injection. 
But, I am not quite sure if it is possible to do that. 
If we can do it in Go, we would love to do it. 
But, I searched for a while, I could not find the proper solution.

> You just need the pdfplumber dependencies in whatever environment you're running go test.

It indeed concerns me.
I treat Python code as a kind of external service, which is similar to API.
So, for me, it is the similar concept to mock API here.

Reply:
>  But, I am not quite sure if it is possible to do that. If we can do it in Go, we would love to do it.

I'm not sure this can be done cleanly, I wouldn't spend too much effort there.

About mocking the command vs running it in the tests, I understand your point in treating it as an API, I'd approach it differently but it isn't a blocker on my side.